### ISSUE TITLE
JSDoc: "export default" prevents type annotation

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/TypeScript#16240]
**TypeScript Version:**  2.3.4

**Code**

``` js
/** @type {React.ComponentClass<any>} */
class Button1 extends React.Component {}

Button1.propTypes = {}; // no error, good!

// However, if we do the same but this time export the class as the default member, we get an unexpected type checking error:

/** @type {React.ComponentClass<any>} */
export default class Button2 extends React.Component {}

Button2.propTypes = {}; // unexpected error!
```

**Comment by DanielRosenwasser** [comment]
Linking back to #24033.
